### PR TITLE
Pull request and issue templates for GitHub

### PR DESCRIPTION
**Pull request by will-in-wi**
I gave a first crack at default issue and PR templates.

Thoughts?

### CODE REVIEW

**Comment by will-in-wi** [comment]
I should give credit. This is based on the Rails templates.

**Comment by mattbrictson** [comment]
Thanks!

We already have a `./issue_template.md` that @leehambley set up, I believe; can you merge your suggestions with that?

I'll have more time to do a full review on Friday.

**Comment by will-in-wi** [comment]
Will do. I'll try and get the improvement done before Friday.

**Comment by leehambley** [comment]
@will-in-wi I believe the only part of mine that I prefer over yours is the part about:

> Important: GitHub issues are for feature requests or bug reports. The Capistrano team recommends you use Stack Overflow for general questions. For more details, please see our contribution policy.

```
**Important:** GitHub issues are for feature requests or bug reports. The Capistrano team recommends you use [Stack Overflow](http://stackoverflow.com/questions/tagged/capistrano) for general questions. For more details, please see our [contribution policy](https://github.com/capistrano/capistrano/blob/master/CONTRIBUTING.md).
```

Otherwise looks good to me

**Comment by will-in-wi** [comment]
Thanks! I've pushed the changes. Let me know what you think.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+Provide a general description of the code changes in your pull
+request... were there any bugs you had fixed? If so, mention them. If
+these bugs have open GitHub issues, be sure to tag them here as well,
+to keep the conversation linked together.
+
+### Short checklist
+
+[ ] Did you run `bundle exec rubocop -a` to fix linter issues?
+[ ] If relevant, did you create a test?
+[ ] Did you confirm that the RSpec tests pass?
```
Comment (mattbrictson) [inline .github/pull_request_template.md]: To get these to appear as actual checkboxes in GitHub, I believe the correct Markdown syntax is:

```
- [ ] 
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+### Other Information
+
+If there's anything else that's important and relevant to your pull
+request, mention that information here.
+
+If you are fixing a bug or introducing a new feature, please add a CHANGELOG entry.
```
Comment (mattbrictson) [inline .github/pull_request_template.md]: How about moving this to the "Short checklist" section? I feel like the CHANGELOG is the most commonly-overlooked step, so it should be prominent.

File: .github/issue_template.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+**Important:** GitHub issues are for feature requests or bug reports. The Capistrano team recommends you use [Stack Overflow](http://stackoverflow.com/questions/tagged/capistrano) for general questions. For more details, please see our [contribution policy](https://github.com/capistrano/capistrano/blob/master/CONTRIBUTING.md).
+
+---
+
+### Steps to reproduce
+If at all possible, please try to create a new Capistrano example project which reproduces the issue you are seeing. You can then upload the individual files to a GitHub Gist or a GitHub project. Often times, an issue is resolved simply by making this test case.
```
Comment (mattbrictson) [inline .github/issue_template.md]: I agree this makes the maintainers' lives easier, but I think this is daunting to ask of someone reporting a bug. You can't really create a  working example Capistrano project without real servers and a `:repo_url` that points to real source code, none of which can be accessed without the reporter's credentials. Unless I am misunderstanding what you are asking for here?

**Comment by mattbrictson** [comment]
@will-in-wi Thanks for putting this together. I added some inline comments. My main concern is that we may discourage people from reporting valid issues if we make the issue template too demanding. The PR template looks excellent and I only suggested some minor edits.

File: .github/issue_template.md (hunk 3 above)
Comment (will-in-wi) [inline .github/issue_template.md]: I understand the concern… I'm mostly thinking of the situation in https://github.com/capistrano/capistrano/issues/1748 where I was able to create https://gist.github.com/will-in-wi/527327e31af30b3eae2068e2965be05b which was the skeleton of a reproducible case.

Perhaps weakening the language would help, e.g. "if at all possible" should probably be "if reasonable" or "if you are running into an issue in Capistrano itself".

Does it make sense what I'm getting at? I'm not sure what the right approach is.

**Comment by will-in-wi** [comment]
Thanks for your comments! I agree with the concern regarding making the template less demanding and have added some inline comments in that vein.

File: .github/issue_template.md (hunk 3 above)
Comment (mattbrictson) [inline .github/issue_template.md]: I think your example Gist does a great job of illustrating what this paragraph is asking for. Could you reference that Gist with a link as part of that text? That would make this ready to ship, IMO.

Comment (will-in-wi) [inline .github/issue_template.md]: I have now done this.

**Comment by mattbrictson** [comment]
Looks ready to merge! Any last-minute edits?

**Pull request merged**